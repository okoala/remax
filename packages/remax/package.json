{
  "name": "remax",
  "version": "1.2.2",
  "description": "Remax 是一个全新的基于 React 的小程序开发框架",
  "main": "cjs/index.js",
  "module": "esm/index.js",
  "keywords": [
    "react",
    "miniapp",
    "mini-program",
    "wechat"
  ],
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/remaxjs/remax/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/remaxjs/remax.git"
  },
  "homepage": "https://remaxjs.org",
  "scripts": {
    "watch": "concurrently \"npm:watch:*\"",
    "watch:esm": "npm run build:esm -- -w",
    "watch:cjs": "npm run build:cjs -- -w",
    "build": "npm run build:esm && npm run build:cjs",
    "build:esm": "tsc",
    "build:cjs": "tsc --module CommonJS --outDir cjs",
    "test": "jest"
  },
  "dependencies": {
    "@remax/mini-types": "^0.0.5",
    "@remax/miniprogram-api-typings": "^2.8.4",
    "lodash.kebabcase": "^4.1.1",
    "react-reconciler": "0.20.4",
    "scheduler": "0.13.6",
    "shallowequal": "^1.1.0"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.2.3",
    "@types/jest": "^24.0.15",
    "@types/lodash.debounce": "^4.0.6",
    "@types/lodash.kebabcase": "^4.1.6",
    "@types/react": "^16.8.22",
    "@types/react-reconciler": "^0.18.0",
    "@types/react-test-renderer": "^16.9.0",
    "@types/shallowequal": "^1.1.1",
    "concurrently": "^5.0.0",
    "jest": "^24.8.0",
    "object-path-immutable": "^3.1.1",
    "react": "^16.8.6",
    "react-test-renderer": "^16.9.0",
    "ts-jest": "^24.0.2",
    "typescript": "^3.5.2",
    "umi": "^2.8.11"
  },
  "optionalDependencies": {
    "antd-mobile": "^2.2.14"
  },
  "gitHead": "28ef4f50ef3c134c4e661d786ebea4d7c4ea89ed"
}
