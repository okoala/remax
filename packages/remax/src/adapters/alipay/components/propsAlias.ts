import plainStyle from '../../../utils/plainStyle';

const alias: any = {
  className: 'class',
  onClick: 'onTap',
  catchClick: 'catchTap',
};

const nativeAlias: any = {
  className: 'class',
};

export function getAlias(prop: string, isNative = false) {
  const aliasProp = isNative ? nativeAlias[prop] : alias[prop];

  if (aliasProp) {
    return aliasProp;
  }

  return prop;
}

function getValue(prop: string, value: any): any {
  if (prop.toLowerCase().endsWith('style') && prop !== 'layerStyle') {
    return plainStyle(value);
  }

  return value;
}

export interface GenericProps {
  [key: string]: any;
}

export default function propsAlias(props: GenericProps, isNative = false) {
  const aliasProps: GenericProps = {};

  Object.keys(props).forEach(prop => {
    if (prop === 'style') {
      aliasProps.style = plainStyle(props.style!);
    } else {
      aliasProps[getAlias(prop, isNative)] = getValue(prop, props[prop]);
    }
  });

  return aliasProps;
}
