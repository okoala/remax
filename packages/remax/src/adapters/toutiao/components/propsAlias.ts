import kebabCase from 'lodash.kebabCase';
import plainStyle from '../../../utils/plainStyle';

const alias: { [prop: string]: string } = {};

export function getAlias(prop: string, isNative = false, isCompile = false) {
  const aliasProp = alias[prop];

  if (aliasProp) {
    return aliasProp;
  }

  if (prop.endsWith('className')) {
    return prop.replace('className', 'class');
  }

  if (prop.startsWith('on')) {
    prop = prop.toLowerCase().replace('on', 'bind');

    if (!isNative) {
      prop = prop.replace('click', 'tap');
    }

    return prop;
  }

  return isCompile ? kebabCase(prop) : prop;
}

function getValue(prop: string, value: any): any {
  if (prop.toLowerCase().endsWith('style') && prop !== 'layerStyle') {
    return plainStyle(value);
  }

  return value;
}

export interface GenericProps {
  [key: string]: any;
}

export default function propsAlias(
  props: GenericProps,
  isNative = false,
  isCompile = false
) {
  const aliasProps: GenericProps = {};

  Object.keys(props).forEach(prop => {
    aliasProps[getAlias(prop, isNative, isCompile)] = getValue(
      prop,
      props[prop]
    );
  });

  return aliasProps;
}
