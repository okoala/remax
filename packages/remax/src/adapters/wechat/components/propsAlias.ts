import kebabCase from 'lodash.kebabCase';
import plainStyle from '../../../utils/plainStyle';

const alias: { [prop: string]: string } = {
  activeColor: 'activeColor',
  backgroundColor: 'backgroundColor',
  onClick: 'bindtap',
  catchClick: 'catchtap',
  enable3D: 'enable-3D',
  hTouchMove: 'htouchmove',
  vTouchMove: 'vtouchmove',
};

export function getAlias(prop: string, isNative = false, isCompile = false) {
  const aliasProp = alias[prop];

  if (aliasProp) {
    return aliasProp;
  }

  if (prop.endsWith('className')) {
    return prop.replace('className', 'class');
  }

  if (prop.startsWith('on') || prop.startsWith('catch')) {
    prop = prop.toLowerCase().replace('on', 'bind');

    if (!isNative) {
      prop = prop.replace('click', 'tap');
    }

    return prop;
  }

  return isCompile ? kebabCase(prop) : prop;
}

function getValue(prop: string, value: any): any {
  if (prop.toLowerCase().endsWith('style') && prop !== 'layerStyle') {
    return plainStyle(value);
  }

  return value;
}

export interface GenericProps {
  [key: string]: any;
}

export default function propsAlias(
  props: GenericProps,
  isNative = false,
  isCompile = false
) {
  const aliasProps: GenericProps = {};

  Object.keys(props).forEach(prop => {
    aliasProps[getAlias(prop, isNative, isCompile)] = getValue(
      prop,
      props[prop]
    );
  });

  return aliasProps;
}
