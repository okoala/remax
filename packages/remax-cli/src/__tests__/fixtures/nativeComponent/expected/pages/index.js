import propsAlias from '../npm/remax/esm/adapters/alipay/components/propsAlias.js';
import '../npm/remax/esm/render.js';
import { createElement } from 'react';
import '../npm/remax/esm/createAppConfig.js';
import createPageConfig from '../npm/remax/esm/createPageConfig.js';
import '../npm/remax/esm/index.js';
import View from '../npm/remax/esm/adapters/alipay/components/View.js';
import '../npm/remax/esm/adapters/alipay/components/ScrollView.js';
import '../npm/remax/esm/adapters/alipay/components/Swiper.js';
import '../npm/remax/esm/adapters/alipay/components/SwiperItem.js';
import '../npm/remax/esm/adapters/alipay/components/MovableView.js';
import '../npm/remax/esm/adapters/alipay/components/MovableArea.js';
import '../npm/remax/esm/adapters/alipay/components/CoverView.js';
import '../npm/remax/esm/adapters/alipay/components/CoverImage.js';
import '../npm/remax/esm/adapters/alipay/components/Icon.js';
import '../npm/remax/esm/adapters/alipay/components/Text.js';
import '../npm/remax/esm/adapters/alipay/components/RichText.js';
import '../npm/remax/esm/adapters/alipay/components/Progress.js';
import '../npm/remax/esm/adapters/alipay/components/Button.js';
import '../npm/remax/esm/adapters/alipay/components/CheckboxGroup.js';
import '../npm/remax/esm/adapters/alipay/components/Checkbox.js';
import '../npm/remax/esm/adapters/alipay/components/Form.js';
import '../npm/remax/esm/adapters/alipay/components/Input.js';
import '../npm/remax/esm/adapters/alipay/components/Label.js';
import '../npm/remax/esm/adapters/alipay/components/Picker.js';
import '../npm/remax/esm/adapters/alipay/components/PickerView.js';
import '../npm/remax/esm/adapters/alipay/components/PickerViewColumn.js';
import '../npm/remax/esm/adapters/alipay/components/RadioGroup.js';
import '../npm/remax/esm/adapters/alipay/components/Radio.js';
import '../npm/remax/esm/adapters/alipay/components/Slider.js';
import '../npm/remax/esm/adapters/alipay/components/Switch.js';
import '../npm/remax/esm/adapters/alipay/components/TextArea.js';
import '../npm/remax/esm/adapters/alipay/components/Navigator.js';
import '../npm/remax/esm/adapters/alipay/components/Image.js';
import '../npm/remax/esm/adapters/alipay/components/Map.js';
import '../npm/remax/esm/adapters/alipay/components/Canvas.js';
import '../npm/remax/esm/adapters/alipay/components/WebView.js';
import '../npm/remax/esm/adapters/alipay/components/Lifestyle.js';
import '../npm/remax/esm/adapters/alipay/components/ContactButton.js';
import '../npm/remax/esm/adapters/alipay/api.js';

var _page = function _page() {
  return createElement(View, null, createElement(A, null), createElement(B, null), createElement(C, null), createElement(D, null), createElement(Complex, null), createElement(PluginComponent, null), createElement(PluginComponent2, null));
};

var index = Page(createPageConfig(_page));var A = function(props) {
          return createElement(
              'a-0',
              propsAlias(props, true),
              props.children
            );
          };var B = function(props) {
          return createElement(
              'b-1',
              propsAlias(props, true),
              props.children
            );
          };var C = function(props) {
          return createElement(
              'c-2',
              propsAlias(props, true),
              props.children
            );
          };var D = function(props) {
          return createElement(
              'd-3',
              propsAlias(props, true),
              props.children
            );
          };var Complex = function(props) {
          return createElement(
              'complex-4',
              propsAlias(props, true),
              props.children
            );
          };var PluginComponent = function(props) {
          return createElement(
              'my-plugin-5',
              propsAlias(props, true),
              props.children
            );
          };var PluginComponent2 = function(props) {
          return createElement(
              'my-plugin-6',
              propsAlias(props, true),
              props.children
            );
          };

export default index;
