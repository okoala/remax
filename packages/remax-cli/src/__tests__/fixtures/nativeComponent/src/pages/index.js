import * as React from 'react';
import { View } from 'remax/alipay';
import A from '../components/a';
import B from '@/components/b';
import C from '../components/c/index';
import D from '../components/d/index.js';
import Complex from '../components/complex';
import PluginComponent from 'plugin://myPlugin/xx';
import PluginComponent2 from 'plugin://myPlugin/dddd';

export default () => (
  <View>
    <A />
    <B />
    <C />
    <D />
    <Complex />
    <PluginComponent />
    <PluginComponent2 />
  </View>
);
