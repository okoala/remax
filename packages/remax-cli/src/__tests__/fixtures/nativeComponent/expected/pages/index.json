{
  "usingComponents": {
    "a-0": "../components/a/index",
    "b-1": "../components/b/index",
    "c-2": "../components/c/index",
    "d-3": "../components/d/index",
    "complex-4": "../components/complex/index",
    "my-plugin-5": "plugin://myPlugin/xx",
    "my-plugin-6": "plugin://myPlugin/dddd"
  }
}